test/unit/models.js

```diff
@@ -0,0 +1,301 @@
+      const quantityBeforeUpdate = 30;
+      const quantityAfterUpdate = 20;
+
+      it('returns object with it\'s properties', async () => {
+        const { _id, quantity } = await productsModel
+          .createProduct(payloadProduct.name, payloadProduct.quantity);
+        expect(quantity).to.equal(quantityBeforeUpdate);
+        const result = await productsModel
+          .updateProduct(payloadUpdate.name, payloadUpdate.quantity, _id);
+        expect(result).to.be.a('object');
+        expect(result).to.have.all.keys('_id', 'name', 'quantity');
+        expect(result.quantity).to.equal(quantityAfterUpdate);
+      });
+    });
+  });
+
+  describe('delete product', () => {
+    describe('when delete is successful', () => {
+      const payloadProduct = {
+        'name': 'rice',
+        'quantity': 30
+      };
+      it('should not exist the product', async () => {
+        const { _id } = await productsModel
+          .createProduct(payloadProduct.name, payloadProduct.quantity);
+        const product = await productsModel.getById(_id);
+        expect(product).to.be.a('object');
+        expect(product).to.have.all.keys('_id', 'name', 'quantity');
+        await productsModel.deleteProduct(_id);
+        const productAfterDelete = await productsModel.getById(_id);
+        expect(productAfterDelete).to.equal(null);
+      });
+    });
+  });
+
+
+  
+});
+
+describe('test sales model', () => {
+  before( async () => {
+    const DBserver = new MongoMemoryServer();
+    const URLMock = await DBserver.getUri();
+  
+    const connectionMock = await MongoClient.connect(URLMock, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true
+    });
+  
+    sinon.stub(MongoClient, 'connect').resolves(connectionMock);
+  
+  });
+  after(() => {
+    MongoClient.connect.restore();
+  });
+
+
+  const payloadProduct =     {
+    'name': 'cookie',
+    'quantity': 10
+  };
+
+  const payloadProduct2 =     {
+    'name': 'candy',
+    'quantity': 66
+  };
+
+  describe('create sale', () => {
+    describe('when creation is successful' ,() => {
+      
+      it('returns an array', async () => {
+        
+        const product1 = await productsModel
+          .createProduct(payloadProduct.name, payloadProduct.quantity);
+
+        const product2 = await productsModel
+          .createProduct(payloadProduct2.name, payloadProduct2.quantity);
+
+
+        const payloadSale = [
+          {
+            '_id': product1._id,
+            'quantity': 2
+          }, 
+          {'_id': product2._id,
+            'quantity': 5}
+        ];
+
+        const result = await salesModel.createSale(payloadSale);
+        expect(result.ops).to.be.a('array');
+        expect(result.ops[0]._id).to.be.be.a('object');
+        expect(result.ops[0].itensSold).to.be.a('array');
+        expect(result.ops[0].itensSold[0]).to.be.a('object');
+        expect(result.ops[0].itensSold[0]).to.have.all.keys('_id', 'quantity');
```

Comment:
Gostei desse modo de fazer os expects. Uma sugestão para evitar um pouca a repetição seria uma desconstrução do result.ops:
```suggestion
        const result = await salesModel.createSale(payloadSale);
        expect(result.ops).to.be.a('array');
        
        const [ {_id, itensSold}] = result;
        expect(_id).to.be.be.a('object');
        expect(itensSold).to.be.a('array');
        expect(itensSold[0]).to.be.a('object');
        expect(itensSold[0]).to.have.all.keys('_id', 'quantity');
```